Sources/System/Process/Signal.swift

```diff
@@ -0,0 +1,114 @@
+
+public struct Signal: RawRepresentable, Hashable {
+  public var rawValue: CInt
+  public init(rawValue: CInt) { self.rawValue = rawValue }
+  fileprivate init(_ rawValue: CInt) { self.init(rawValue: rawValue) }
+}
+
+// FIXME(DO NOT MERGE): Migrate these to the constants.swift file
+import Darwin
+
+extension Signal {
+  #if os(Linux)
+  public static var unused: Singal { Signal(SIGUNUSED) }
+  #endif
+
+  // TODO: better names
+
+  /// SIGHUP (1): terminal line hangup (default behavior: terminate process)
+  public static var hangup: Signal { Signal(SIGHUP) }
+
+  /// SIGINT (2): interrupt program (default behavior: terminate process)
+  public static var interrupt: Signal { Signal(SIGINT) }
+
+  /// SIGQUIT (3): quit program (default behavior: create core image)
+  public static var quit: Signal { Signal(SIGQUIT) }
+
+  /// SIGILL (4): illegal instruction (default behavior: create core image)
+  public static var illegalInsruction: Signal { Signal(SIGILL) }
+
+  /// SIGTRAP (5): trace trap (default behavior: create core image)
+  public static var traceTrap: Signal { Signal(SIGTRAP) }
+
+  /// SIGABRT (6): abort program (formerly SIGIOT) (default behavior: create core image)
+  public static var abort: Signal { Signal(SIGABRT) }
+
+  /// SIGEMT (7): emulate instruction executed (default behavior: create core image)
+  public static var emulatorTrap: Signal { Signal(SIGEMT) }
+
+  /// SIGFPE (8): floating-point exception (default behavior: create core image)
+  public static var floatingPointException: Signal { Signal(SIGFPE) }
+
+  /// SIGKILL (9): kill program (default behavior: terminate process)
+  public static var kill: Signal { Signal(SIGKILL) }
+
+  /// SIGBUS (10): bus error (default behavior: create core image)
+  public static var busError: Signal { Signal(SIGBUS) }
+
+  /// SIGSEGV (11): segmentation violation (default behavior: create core image)
+  public static var segfault: Signal { Signal(SIGSEGV) }
```

Comment:
not super how I feel about this. We're not using segmentation anymore but paging, so spelling "segfault" out may not make it easier to find?

Reply:
Having an unavailable-renamed entry for `SIGSEGV` would hopefully help point to whatever name we choose.

Reply:
That's true. But `segmentationViolation` is still not what this will be sent for so I'd be inclined to leave the as a "term of art".

Reply:
Yeah, this is sort of like `hangup` in that the name has somehow outgrown its original meaning, which is now a complete anachronism. SIGSEGV is defined as "segmentation violation" in the man pages, so at least this expansion isn't likely to confuse anyone.

@weissi, what would you propose as a replacement? Note that System is supposed to follow the Swift naming conventions, so the original arbitrary abbreviations need to be replaced, no matter how well known.

Reply:
FWIW, [POSIX describes SIGSEGV](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/signal.h.html#tag_13_42) as "Invalid memory reference". Would we be willing to name it `invalidMemoryReference`?

SEGV is such an (in)famous signal, it may be worth spending a term-of-art card on `segmentationViolation`, no matter how far it is from the current meaning...

Reply:
`invalidMemoryReference` is IMHO much better than `segmentationViolation`. Both names however aren't as good as `SIGSEGV` or `sigsegv` IMHO. If you want to program your operating system directly, I think any translation of names is counterproductive and will lead to avoidable bugs, imprecise reviews, etc.